### ISSUE TITLE
urllib3 1.25 breaks ssl verification when verify_certs=False

### ISSUE THREAD

**Issue by jrobison-sb** [elastic/elasticsearch-py#943]
Our code:

```
Elasticsearch(os.environ['ELASTICSEARCH_URL'],
                             verify_certs=False, timeout=60)
```

When our underlying pip3 package of `urllib3` is at 1.24.2, the above works great.

When we use `urllib3` 1.25, it breaks with the following error:
```
ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:720)
```

Considering that urllib3 [changed the way that SSL verification works by default](https://github.com/urllib3/urllib3/blob/master/CHANGES.rst) in 1.25, I suspect that either Elasticsearch isn't correctly passing `verify_certs=False` to urllib3, or urllib3 isn't honoring it, or something...

I don't know for sure if this is an elasticsearch-py problem or a urllib3 problem, but since my goal is to use Elasticsearch, I figured I should start here. Thanks.

OS: Debian Stretch, in a docker container
Python: 3.5.3
Elasticsearch python package: 7.0.0
Elasticsearch server: 6.2, using AWS's managed Elasticsearch service

**Comment by vtermanis** [comment]
Shameless bump - hopefully addressed by #987

**Comment by jolyj** [comment]
I had the problem here, the only fix was to downgrade to 1.24.2 to make urllib3 ignore certificate verification

**Comment by xNinjaKittyx** [comment]
I had this problem with requests library. It completely ignored my verify=False parameter in requests.

I was on 1.25.4, but had to revert to 1.25.3 for it to work properly again.

**Comment by mlcdf** [comment]
> I had the problem here, the only fix was to downgrade to 1.24.2 to make urllib3 ignore certificate verification

Thank for the workaround!

**Comment by vtermanis** [comment]
@HonzaKral - is [the fix](https://github.com/elastic/elasticsearch-py/commit/4c531c5cdbd5ba8b59be16c2fa15d5cf81d0a92f) going to be backported to the ES6 branch of this package?

**Comment by mccarthyp-snet** [comment]
I downgraded urllib3 to 1.24.2 and I'm still getting `sslv3 alert bad certificate`.

Python: 3.8.0
Python client for Elasticsearch: 7.6.0
Elasticsearch: 6.2.3

**Comment by sanand0** [comment]
Note: [urllib3 1.26.4 and below have a security vulnerability](https://snyk.io/vuln/SNYK-PYTHON-URLLIB3-1533435). I believe we'll need to upgrade to 1.26.5 or above.

**Comment by sethmlarson** [comment]
The security vulnerability mentioned above wouldn't impact almost all users of this library (would have to use user input as a URL to Elasticsearch, very unlikely) so using 1.25 should be fine. As a work-around for this issue either use `verify_certs=False` with 1.25, upgrade urllib3 to 1.26 and use `verify_certs=True`. In Elasticsearch 8.0 and 7.x the issue is resolved as urllib3 1.26+ is required.
